Fallback to Get when Head Fails

joaopapereira opened the pull request:
Related to #131

joaopapereira committed:
Fallback to Get when Head Fails

In this commit we are falling back to a Get Request when the Head
request done to retrieve the Digest fails. This is done to prevent some
issues that we found with some registries when doing a HEAD Request,
where they were returning a 401 even for public images.
We decided to keep the Head request, because it is more performant and
will be sucessful in the majority of the cases.
This also helps due to some rate limitation that can happen when doing
Gets

cppforlife commented on pkg/imgpkg/registry/registry.go:
```diff
@@ -97,7 +97,11 @@ func (r Registry) Digest(ref regname.Reference) (regv1.Hash, error) {
 	desc, err := regremote.Head(overriddenRef, r.opts...)
 	if err != nil {
-		return regv1.Hash{}, err
+		getDesc, err := r.Get(overriddenRef)
```
may be use `regremote.Get` here given ambiguity of r.Get vs r.Generic?

joaopapereira committed:
Correct Get logic

Looks like the Get logic was missing the refOpts when parsing the
reference of the image.
Remove the Generic call since it can be done by calling Get
